myArray.js

```diff
@@ -0,0 +1,84 @@
+"use strict";
+
+var removeDuplicates = require('./duplicates');
+var findMostFrequent = require('./frequency');
+
+Array.prototype.myPush = function(item) {
+
+  // Add item to end of array ("Push")
+  this[this.length] = item;
+
+  return this.length;
+};
+
+Array.prototype.myPop = function() {
+
+  // Saved item to be removed ("pop")
+  var poppedItem = this[this.length-1];
+
+  // Remove last item ("pop")
+  this.length--;
+
+  return poppedItem;
+};
+
+Array.prototype.myShift = function() {
+
+  // Save 1st item to be removed
+  var shiftedItem = this[0];
+
+  // Remove 1st item by copying all items
+  // down one index in array starting with
+  // 2nd item
+  for (var i = 1; i < this.length; i++) {
+    this[i-1] = this[i];
+  }
+
+  this.length--;
+
+  // If array is not empty, return the shifted
+  // item. If array is empty, return undefined
+  if (this.length) {
+    return shiftedItem;
+  } else {
+    return undefined;
+  }
+};
```

Comment:
Shouldn't this just be `return shiftedItem;`? Otherwise if you have array of `[ 'foo' ]` and you call `myShift` then would return `undefined`